lib/commands/settings.js

```diff
@@ -45,4 +45,33 @@ commands.setTimeDilation = async function (factor) {
   return result.value;
 };
 
+
+// Sets the attribute type/value to filter the source tree
+// format: 
+// SourceTreeFilter: "[@attributeType='attributeValue']"
+commands.setSourceTreeFilter = async function (filter) {
+  var commandObject = {
+    name: `setSourceTreeFilter`,
+    args: [`${filter}`]
+  };
+  var commandJSON = JSON.stringify(commandObject);
+  let data = await this.executeSocketCommand(commandJSON);
+  
+  let result;
+  try {
+    result = JSON.parse(data);
+  } catch (e) {
+    throw new Error("Bad response from setValue");
+  }
+  
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_UNKNOWN_COMMAND) {
+    throw new errors.UnknownCommandError('argument must be numeric and larger than zero');
+  }
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_INVALID_SELECTOR) {
+    throw new errors.UnknownCommandError("Attribute filter should have following format: [@attributeType='attributeValue']");
```

Comment:
Shall we not through an `InvalidSelectorError` here instead [Selenium Errors](http://www.rubydoc.info/gems/selenium-webdriver/Selenium/WebDriver/Error/InvalidSelectorError)

Reply:
yes